core/src/value/primitive.rs

```diff
@@ -965,6 +965,7 @@ impl PrimitiveValue {
         match self {
             PrimitiveValue::Str(s) => {
                 s.trim_end()
+                    .trim_start()
```

Comment:
is there a reason to not go with [trim](https://doc.rust-lang.org/std/string/struct.String.html#method.trim) everywhere?

Reply:
Hi @ingwinlu ,
DICOM VR definition"Decimal Strings may be padded with leading or trailing spaces. Embedded spaces are not allowed." 
So it depends if you want to raise an error with embedded spaces or not?

Reply:
Updated the code 👍